yt/utilities/lib/cyoctree.pyx

```diff
@@ -0,0 +1,412 @@
+cimport numpy as np
+import numpy as np
+cimport cython
+
+# for writing to file
+import struct
+
+# cpp includes
+from libcpp.vector cimport vector
+from libcpp cimport bool
+
+# c includes
+cimport libc.math as math
+
+from yt.geometry.particle_deposit cimport \
+    kernel_func, get_kernel_func
+
+cdef struct Node:
+    double left_edge[3]
+    double right_edge[3]  # may be more efficient to store the width instead
+    int start             # which particles we store
+    int end
+    int parent            # position of parent in Octree.nodes
+    int children          # position of 0th child, children are contiguous
+    bool leaf
+    unsigned int node_id  # these probably should be longs
+    unsigned int leaf_id
+    unsigned short depth
+
+cdef struct Octree:
```

Comment:
Yes, and it probably makes sense to do so. Originally, there was no `PyOctree` class and it was just this node when I started writing. But when I came to wire it in, the `PyOctree` class felt very natural. I think this would be a lot more natural without this, thank you

Reply:
fixed